FlagsmithClient/Classes/Internal/SSEManager.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  SSEManager.swift
+//  FlagsmithClient
+//
+//  Created by Gareth Reese on 13/09/2024.
+//
+
+import Foundation
+#if canImport(FoundationNetworking)
+    import FoundationNetworking
+#endif
+
+/// SSEManager handles interaction with the Flagsmith SSE real-time API.
+/// It manages the connection to the SSE endpoint, processes incoming events,
+/// and handles reconnection logic with a backoff strategy.
+final class SSEManager: NSObject, URLSessionDataDelegate, @unchecked Sendable {
+    private var _session: URLSession!
+    private var session: URLSession {
+        get {
+            propertiesSerialAccessQueue.sync { _session }
+        }
+        set {
+            propertiesSerialAccessQueue.sync(flags: .barrier) {
+                _session = newValue
+            }
+        }
+    }
+
+    private var _dataTask: URLSessionDataTask?
+    private var dataTask: URLSessionDataTask? {
+        get {
+            propertiesSerialAccessQueue.sync { _dataTask }
+        }
+        set {
+            propertiesSerialAccessQueue.sync(flags: .barrier) {
+                _dataTask = newValue
+            }
+        }
+    }
+
+    /// Base `URL` used for requests.
+    private var _baseURL = URL(string: "https://realtime.flagsmith.com/")!
+    var baseURL: URL {
+        get {
+            propertiesSerialAccessQueue.sync { _baseURL }
+        }
+        set {
+            propertiesSerialAccessQueue.sync {
+                _baseURL = newValue
+            }
+        }
+    }
+
+    /// API Key unique to an organization.
```

Comment:
Probably we mean `environment` here instead of organisation. Also note that we defiantly use British english throughout the Flagsmith project ;)

Reply:
Have updated throughout